### PR TITLE
SUP-1518 Run snapshots on buildkite

### PR DESCRIPTION
**Pull request by jradtilbrook**

### CODE REVIEW

File: app/examples/github-actions/laravel-tests.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        php: [8.1, 8.2, 8.3]
+        php: ['8.1', '8.2', '8.3']
```
Comment (toote) [inline app/examples/github-actions/laravel-tests.yml:42]: What is the reason for this change? If it is that the generated BK pipeline needs them to be strings, we should have a bugfix on the translation to ensure they are always strings instead of changing them here as this would still be a valid GHA configuration we might find

Comment (jradtilbrook) [reply app/examples/github-actions/laravel-tests.yml:42]: just a temporary fix for now to get the build green, ill work on ensuring its all strings soon

Comment (pzeballos) [reply app/examples/github-actions/laravel-tests.yml:42]: I don't think is a good idea to change tests in order to have green builds, that could bring lots of problems, not to mention that it is bad practice. The bugfix doesn't seem to be very complicated 🤔

Comment (jradtilbrook) [reply app/examples/github-actions/laravel-tests.yml]: hmm so a problem I'm having is that I can map the values to strings. But when its printed back to yaml, it seems to convert it back to numeric values again. I'm not sure why thats happening. See the latest commit for the mapping, but the output is still numeric

File: app/lib/bk/compat/parsers/gha/matrix.rb [hunk 2]
```diff
@@ -5,6 +5,7 @@ module Compat
     class GitHubActions
       def set_matrix(bk_step, config)
         matrix = config['strategy']['matrix']
+        matrix.map { |key, value| [key, value.map(&:to_s)] }.to_h
```
Comment (toote) [inline app/lib/bk/compat/parsers/gha/matrix.rb:8]: `map` returns a copy of the array, you can ensure all values are strings with:

```suggestion
        matrix = config['strategy']['matrix'].transform_values(&:to_s)
```

File: app/examples/github-actions/laravel-tests.yml (hunk 1 above)
Comment (toote) [reply app/examples/github-actions/laravel-tests.yml]: left a comment with an explanation and way to solve the issue

File: app/lib/bk/compat/parsers/gha/matrix.rb (hunk 2 above)
Comment (jradtilbrook) [reply app/lib/bk/compat/parsers/gha/matrix.rb]: Oh damn that has changed the whole array to a string so its not quite what we want

Comment (toote) [reply app/lib/bk/compat/parsers/gha/matrix.rb:8]: Oh... in that case I believe the solution would be have something like [the `recurse_to_string` method we use for transforming steps](https://github.com/buildkite/migration/blob/main/app/lib/bk/compat/pipeline/step.rb#L169-L181) applied here

**Comment by jradtilbrook** [comment]
Alright, got this working with some pairing with @toote

File: app/lib/bk/compat/parsers/gha/matrix.rb [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
+        case value
+        when Hash
+          value.transform_values! { |elem| convert_to_string(elem) }
+        when Array
+          value.map! { |elem| convert_to_string(elem) }
+        when Numeric
+          "#{value}"
+        else
+          # if we don't know how to do this, do nothing
+          value
```
Comment (toote) [inline app/lib/bk/compat/parsers/gha/matrix.rb:26]: why not just?
```suggestion
        else
          value.to_s
```

**Review by toote — APPROVED** [review]
